### DISCUSSION TITLE
Old workflow used after commit change?

### DISCUSSION THREAD

**Discussion by pcolmer** [community/community#25437] [Actions]
<p>I’m using <code>pull_request_target</code> to trigger a workflow when a PR is opened and updated. In my testing, I’ve created a PR with <code>develop</code> as the updated branch and <code>master</code> as the branch I want to merge into.</p>
<p>Since I’m using <code>pull_request_target</code>, my understanding is that Actions will use the workflow from the <code>master</code> branch. I’m still building up my workflow bit by bit and fixing the bugs in it. However, what I’m finding is that if I commit the revised workflow back to the <code>master</code> branch and then update the PR so that the action is triggered, it often seems like the <em>previous</em> version of the workflow is used and not the version I’ve just committed.</p>
<p>Am I doing something wrong or am I misunderstanding something about how Actions work?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/pcolmer">@pcolmer</a>,</p>
<blockquote>
<p>Since I’m using <strong>pull_request_target</strong>, my understanding is that Actions will use the workflow from the <strong>master</strong> branch.</p>
</blockquote>
<p>Yes, your understanding is correct.<br>
However, there are more things you also need to know:</p>
<ol>
<li>
<p>When opening a PR, the base commit SHA (<strong>github.event.pull_request.base.sha</strong>) generally will point to the latest commit on the target branch of the PR, if the source branch is up-to-date with the target branch. The base commit SHA normally is fixed and not automatically changed.</p>
</li>
<li>
<p>During the PR is <strong>Open</strong>, if you push some new commits to the target branch of the PR, the base commit SHA also will not be changed automatically. At this time, if you want the base commit SHA can point to the latest commit on the target branch, you can do like as this:</p>
<ul>
<li>If you have set the <strong>Branch protection rule</strong> for the target branch (<strong>master</strong> in your case) and enabled the option “<strong>Require branches to be up to date before merging</strong>”, you will see the “<strong>Update branch</strong>” button on the PR. Click the button to sync the new commits from the target branch to the source branch and the PR merge branch. After that, the base commit SHA will point to the latest commit on the target branch.<br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/a/ba172324d6041dd49dac04462db8defde7f544b8.png" data-download-href="/uploads/short-url/qyenCaUzPrDNCDW0W7s8U7DP7Ys.png?dl=1" title="branch-rule" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/46307996/181091128-a2df2565-cc5b-4925-9495-48d613512e1e.png" alt="branch-rule" data-base62-sha1="qyenCaUzPrDNCDW0W7s8U7DP7Ys" width="690" height="423" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/a/ba172324d6041dd49dac04462db8defde7f544b8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">branch-rule</span><span class="informations">913×561 30.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</li>
</ul>
<p></p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/f/dfea48badc2638555dffc4daabcd2772504c6ea0.png" data-download-href="/uploads/short-url/vWQquthY9L1GXAAQv5Sx66qVWQ8.png?dl=1" title="update-branch" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/46307996/181091129-394fa683-81c2-48bd-85a3-90c6a3c14889.png" alt="update-branch" data-base62-sha1="vWQquthY9L1GXAAQv5Sx66qVWQ8" width="690" height="192" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/f/dfea48badc2638555dffc4daabcd2772504c6ea0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">update-branch</span><span class="informations">916×256 15.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<ul>
<li>If you do not set the <strong>Branch protection rule</strong>, you need to run the “<a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#update-a-pull-request-branch" rel="noopener nofollow ugc"><strong>Update a pull request branch</strong></a>” API.</li>
</ul>
</li>
</ol>

**Comment by pcolmer** [comment]
<p>Brilliant! Thank you for that explanation and the solutions offered. It makes sense to me now :slight_smile:</p>
